cukes-sql/src/main/java/lv/ctco/cukes/sql/GenericTableRepository.java

```diff
@@ -0,0 +1,72 @@
+package lv.ctco.cukes.sql;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import lombok.SneakyThrows;
+import lv.ctco.cukes.core.CukesRuntimeException;
+import org.apache.commons.lang3.StringUtils;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Singleton
+public class GenericTableRepository {
+
+    private static final String ROW_COUNT_COLUMN = "ROW_COUNT";
+    @Inject
+	private ConnectionFactory connectionFactory;
+
+	@SneakyThrows(SQLException.class)
+	public List<Map<String, String>> getTableValues(String schema, String tableName, List<String> columns) {
+		if (columns.isEmpty()) {
+			throw new CukesRuntimeException("Can not build query from empty columns list");
+		}
+		String query = createSelectQuery(schema, tableName, columns);
+		return tryExecuteQuery(columns, query);
+	}
+
+    @SneakyThrows(SQLException.class)
+    public Integer countTableValues(String schema, String tableName) {
+        String query = createCountQuery(schema, tableName);
+        return Integer.valueOf(tryExecuteQuery(Collections.singletonList(ROW_COUNT_COLUMN), query).get(0).get(ROW_COUNT_COLUMN));
+    }
+
+	private List<Map<String, String>> tryExecuteQuery(List<String> columns, String query) throws SQLException {
+        try (Connection connection = connectionFactory.getConnection();
+             Statement statement = connection.createStatement();
+             ResultSet resultSet = statement.executeQuery(query)) {
```

Comment:
ResultSet should be closed as well

Reply:
all fine here: connection, statement and resultSet are closed in Try-with-resources